### PR TITLE
feat(backend): Yandex cloud lockbox backend

### PR DESCRIPTION
**Pull request by KSerik**
### Description
Add support for Yandex Cloud lockbox secret manager.

### Checklist
Please make sure that your PR fulfills the following requirements:
- [x] Reviewed the guidelines for contributing to this repository
- [x] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] Tests for the changes have been updated
- [x] Docs have been added / updated
- [ ] Optional. My organization is added to USERS.md.

### Type of Change

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### CODE REVIEW

**Review by werne2j — APPROVED** [review]
Thanks @KSerik! Looks good to me, will let @jkayani give his review before merging.

File: docs/backends.md [hunk 1]
```diff
@@ -456,4 +456,59 @@ metadata:
...
+```
+
+### Yandex Cloud Lockbox
+##### YCL Authentication
+Refer to the [IAM overview](https://cloud.yandex.com/en/docs/iam/concepts/) for yandex cloud APIs authorization.
+
+These are the parameters for YCL:
+```
+AVP_TYPE: yandexcloudlockbox
+AVP_YCL_SERVICE_ACCOUNT_ID: Service account ID
```
Comment (jkayani) [inline docs/backends.md:468]: Just one thing to mention - if there is a standard set of env variables that CLI tools (or the go module being used) for Yandex cloud expect to use, I'd prefer that we document those are accepted for AVP and read them in using the code here: https://github.com/argoproj-labs/argocd-vault-plugin/blob/main/pkg/config/config.go#L42 and https://github.com/argoproj-labs/argocd-vault-plugin/blob/main/pkg/config/config.go#L219. The thinking is we want AVP to follow the conventions of other tools in a backends ecosystem to make it predictable. 

But if not, this is perfectly acceptable.

**Review by jkayani — APPROVED** [review]
LGTM, thanks for the code, tests and docs @KSerik. Just the one note for you to follow-up on if applicable.

File: docs/backends.md (hunk 1 above)
Comment (KSerik) [reply docs/backends.md:468]: Sorry guys, I've thoroughly researched the documentation, there is no standard set of env variables for CLI tools and Go-SDK.

**Comment by werne2j** [comment]
@KSerik could you rebase and squash your commits down to one? When that is done, i will merge it in.

**Comment by KSerik** [comment]
> @KSerik could you rebase and squash your commits down to one? When that is done, i will merge it in.

@werne2j Done)

**Comment by werne2j** [comment]
This backend is now available in https://github.com/argoproj-labs/argocd-vault-plugin/releases/tag/v1.9.0

**Pull request merged**